packages/plugin-awesome/test/path-joining.test.ts

```diff
@@ -0,0 +1,136 @@
+import type { PluginContext } from "@allurereport/plugin-api";
+import { describe, expect, it } from "vitest";
+import AwesomePlugin from "../src/index.js";
+import { fixtures, createFilesCollectorWithCheck } from "./fixtures.js";
+import { join as joinPosix } from "node:path/posix";
+
+// Path joining mechanism tests
+
+describe("Path joining mechanism", () => {
+  it("should use posix path joining for all file paths", async () => {
+    const { files, context } = createFilesCollectorWithCheck((normalizedPath) => {
```

Comment:
The test actually checks nothing because we're testing a test function abstraction, not a runtime code.

See [an example in a comment above](https://github.com/allure-framework/allure3/pull/175/files#diff-d61929ef1b174abbba7bd65ea6263e1d72704409117042c277c43c81c9b1b9bdR187) how you can test the readers' functionality in the plugin